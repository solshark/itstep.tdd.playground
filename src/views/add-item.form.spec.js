import { AddItemForm } from "./add-item.form";
let form;

beforeEach(() => {
  form = new AddItemForm();
  document.body.innerHTML = "";
  document.body.append(form.dom);
});
describe("AddItemForm", () => {
  it("should match snaphot", () => {
    expect.assertions(1);

    expect(form.dom).toMatchSnapshot();
  });

  describe("Layout", () => {
    it("should render div#container", () => {
      expect.assertions(1);

      const container = document.getElementById("container");
      expect(container).toBeTruthy();
    });
    it("should render h3#header", () => {
      expect.assertions(1);
      const header = document.getElementById("header");
      expect(header).toBeTruthy();
    });
    it("should render input for title", () => {
      expect.assertions(1);
      const input = document.getElementById("inpTitle");
      expect(input).toBeTruthy();
    });
    it("should render button", () => {
      expect.assertions(1);
      const button = document.getElementById("btnAdd");
      expect(button).toBeTruthy();
    });
  });
  describe("btnAdd", () => {
    it("should be disabled initially", () => {
      expect.assertions(1);
      const button = document.getElementById("btnAdd");
      expect(button.disabled).toBe(true);
    });
    it("should call onItemAdded() method if clicked", () => {
      expect.assertions(1);
      document.body.innerHTML = "";

      const clickSpy = jest
        .spyOn(AddItemForm.prototype, "onAddItemClicked")
        .mockImplementation(() => {
          return true;
        });
      const f = new AddItemForm();
      document.body.append(f.dom);
      const button = document.getElementById("btnAdd");
      button.disabled = false;
      button.click();
      expect(clickSpy).toBeCalled();
    });
  });
});
