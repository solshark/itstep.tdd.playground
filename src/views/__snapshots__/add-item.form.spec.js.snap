// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemForm should match snaphot 1`] = `
<div
  id="container"
>
  Add new item you want to purchase
  <h3
    id="header"
  >
    Add item
  </h3>
  <input
    id="inpTitle"
    placeholder="Type item title here"
  />
  <button
    disabled=""
    id="btnAdd"
    value="Add item"
  />
</div>
`;
